File: Haskell/ArrayStack.hs
module ArrayStack (ArrayStack,top,pop,push,isEmpty,showArrayStack) where

-- This module is an implementation of ArrayStack using my own functions

import Data.Bool

-- type synonym for a ArrayStack
type ArrayStack = [Char]

top :: ArrayStack -> Char
top (a:as) = a

pop :: ArrayStack -> ArrayStack
pop (a:as) = as

push :: Char -> ArrayStack -> ArrayStack
push c s = c : s

isEmpty :: ArrayStack -> Bool
isEmpty [] = True
isEmpty _ = False

showArrayStack :: ArrayStack -> String
showArrayStack [] = ""
showArrayStack (s:ss) = s : showArrayStack ss

File: Java/ArrayStack.java
import java.io.*;
import java.lang.*;
import java.lang.ArrayIndexOutOfBoundsException;
import java.lang.Character;
import java.lang.Exception;
import java.lang.System;
import java.lang.Throwable;
import java.util.ArrayList;

public class ArrayStack implements Stack {

    ArrayList<Character> stack = new ArrayList<Character>();

    public char top() {
        return stack.get(0);
    }

    public void pop(){
        stack.remove(0);
    }

    public void push(char c){
        stack.add(c);
    }

    public boolean isEmpty() {
        if (stack.size() == 0) return true;
        else return false;
    }

    private static PrintStream ps = new PrintStream(System.out);

    public void show(PrintStream p){
        ps.println(stack);
    }
}